alphalens/utils.py

```diff
@@ -205,24 +211,30 @@ def compute_forward_returns(factor_idx,
     Returns
     -------
     forward_returns : pd.DataFrame - MultiIndex
-        Forward returns in indexed by date and asset.
+        Forward returns indexed by date and asset.
         Separate column for each forward return window.
     """
 
-    factor_idx = factor_idx.intersection(prices.index)
+    factor_dateindex = factor.index.levels[0]
+    if factor_dateindex.tz != prices.index.tz:
+        raise NonMatchingTimezoneError("The timezone of 'factor' is not the "
+                                       "same as the timezone of 'prices'. See "
+                                       "the pandas methods tz_localize and "
+                                       "tz_convert.")
 
+    factor_dateindex = factor_dateindex.intersection(prices.index)
+    
     forward_returns = pd.DataFrame(index=pd.MultiIndex.from_product(
-        [factor_idx, prices.columns], names=['date', 'asset']))
+        [factor_dateindex, prices.columns], names=['date', 'asset']))
 
-    freq = infer_trading_calendar(factor_idx, prices.index)
+    freq = infer_trading_calendar(factor_dateindex, prices.index)
```

Comment:
Could you move` infer_trading_calendar` above, before factor date index is changed? Thanks for finding the bug